keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/nuts/NUTS.java

```diff
@@ -114,30 +118,29 @@ private NUTSSampler setupSampler(final ProbabilisticModelWithGradient model,
 
         Map<VariableReference, ?> startingSample = SamplingAlgorithm.takeSample(sampleFromVariables);
 
-        double startingStepSize = (initialStepSize == null) ? Stepsize.findStartingStepSize(
-            position,
-            gradient,
-            latentVariables,
-            model,
-            initialLogOfMasterP,
-            random
-        ) : initialStepSize;
+        double startingStepSize = (initialStepSize == null) ?
+            findStartingStepSizeSimple(0.25, latentVariables) :
+            initialStepSize;
 
-        Stepsize stepsize = new Stepsize(
+        AdaptiveStepSize stepSize = new AdaptiveStepSize(
             startingStepSize,
             targetAcceptanceProb,
             adaptCount
         );
 
-        Tree tree = Tree.createInitialTree(position, new HashMap<>(), gradient, initialLogOfMasterP, startingSample);
+        Potential potential = new AdaptiveQuadraticPotential(zeros(position), ones(position), 10.0, adaptCount, potentialAdaptWindowSize, random);
```

Comment:
I've made Potential a parameter that has a default. It's mean 0, variance 1 just because you have to start somewhere and it is initial weight 10 so that when the sampling first starts it'll take larger  weight to the first couple of samples and adapt faster.